[New Entry/ Edits] Python Builtin-Functions: `super()`/ Inheritance (#1008)
Codecademy/docs: content/python/concepts/built-in-functions/terms/super/super.md

---
Title: 'super()'

> Description: 'implicitly refer to the superclass'

Reviewer: ```suggestion
Description: 'Returns a temporary object that allows a given class to inherit the methods and properties of a parent or sibling class.'
```

Let's make sure to use capitalization and punctuation. 😄

Author: Updated

Subjects:

>   - 'Computer Science'

Reviewer: ```suggestion
  - 'Computer Science'
  - 'Data Science'
```

Since Python is commonly used in data science.

Author: I've added Data Science as a tag. I've actually never encountered super() as a data science term that why I didn't put it in, but I admit, I haven't taken Codecademy's Data Science course.

Reviewer: Yeah these are relatively loose associations meant to bolster the entry's SEO.

Tags:

>   - 'Functions'
>   - 'Objects'
>   - 'Inheritance'

Reviewer: ```suggestion
  - 'Classes'
  - 'Functions'
  - 'Inheritance'
  - 'Methods'
  - 'Objects'
  - 'OOP'
```

Author: Absolutely! Added.

CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> The `super()` function is built into python and is used to refer to the the superclass of an object without specifically using the parent name.

Reviewer: ```suggestion
The **`super()`** function returns a temporary object that allows a given [class](https://www.codecademy.com/resources/docs/python/classes) to [inherit](https://www.codecademy.com/resources/docs/python/inheritance) the methods and properties of a parent or sibling class.
```

Author: I've made this modification.

## Syntax

```pseudo
super().function()
super(type) 
```

## `super()`

`super()` returns a temporary object of the superclass, allowing the superclass’s methods to be called. This comes in handy when using [multiple inheritance](#multiple-inheritance).

> ## Example

Reviewer: ```suggestion
## Codebyte Example

```

Let's allow one empty line above and below each header.

Reviewer: Also let's rename this to "Codebyte Example" since Python is a supported language for [runnable code snippets](https://help.codecademy.com/hc/en-us/articles/1260806285369-Codebytes) (which is compatible with Codecademy Docs 😄)

Author: Updated

In this case, the `.say_hi()` method is implemented utilizing `super()` inside `.intro()`:

> ```py

Reviewer: ```suggestion
```codebyte/python
```

Author: Updated!

> 

Author: Updated "function" to "method" and put a line to indicate what the things are. Not sure if it is a good explanation, trying to stay focused on super without bleeding into other dense topics.

> 

Author: I've moved the text to be below the pseudo section

# Base class
class ProgramLanguage:
  def say_hi(self):
    print("Hi! I am a Programming Language")

# Derived class
class Python(ProgramLanguage):
  def intro(self):
    super().say_hi()
    print("Python here!")

doc = Python()
doc.intro()
```

> 
> The output would be:
> 
> ```shell
> Hi! I am a Programming Language.
> Python here!
> ```

Reviewer: ```suggestion
```

If we choose to go with a Codebyte example, then these lines can be omitted; the output will appear on the screen after the code is run.

Author: I've removed this portion of the doc.

---
Title: 'super()'
Description: 'Returns a temporary object that allows a given class to inherit the methods and properties of a parent or sibling class.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Objects'
  - 'Inheritance'
  - 'Classes'
  - 'Functions'
  - 'Inheritance'
  - 'Methods'
  - 'Objects'
  - 'OOP'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

The **`super()`** function returns a temporary object that allows a given [class](https://www.codecademy.com/resources/docs/python/classes) to [inherit](https://www.codecademy.com/resources/docs/python/inheritance) the methods and properties of a parent or sibling class.

## Syntax

```pseudo
super().method()

> super(type)

Reviewer 2: Can we show how the `super(type)` is used in its own example?

```

To access a method of a parent class or specify the type object of the parent class using pseudocode.

## `super()`

`super()` returns a temporary object of the superclass, allowing the superclass’s methods to be called. This comes in handy when using multiple inheritance to derive a class from more than one base class.

> 

Reviewer 2: Add ## Example subheading. Give an example and brief explanation showing how `super(type)` syntax works.

Author: I made the update to super(type) by updating it to super(type, object) and creating an additional codebyte example for it.

---
Title: 'super()'
Description: 'Returns a temporary object that allows a given class to inherit the methods and properties of a parent or sibling class.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Objects'
  - 'Inheritance'
  - 'Classes'
  - 'Functions'
  - 'Inheritance'
  - 'Methods'
  - 'Objects'
  - 'OOP'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

The **`super()`** function returns a temporary object that allows a given [class](https://www.codecademy.com/resources/docs/python/classes) to [inherit](https://www.codecademy.com/resources/docs/python/inheritance) the methods and properties of a parent or sibling class.

## Syntax

```pseudo
super().method()

> super(type, object)

Author: Fixed super().method() to super()

```

`.method()` is the accessed method of the parent class. `type` specifies the type object of the parent class and object is an instance or subtype of the `type`.

> 

Reviewer: Can we place a second \```pseudo snippet here that looks like this:

```pseudo
class A():
  method(arg):
    # Method code starts here

class B(A):
  method(arg):
    super().method(arg)
    # Method code starts here
```

Author: Updated.
